core/trino-main/src/test/java/io/trino/sql/rewrite/TestStatementRewrite.java

```diff
@@ -0,0 +1,210 @@
+import java.util.function.Consumer;
+
+import static io.trino.connector.CatalogName.createInformationSchemaCatalogName;
+import static io.trino.connector.CatalogName.createSystemTablesCatalogName;
+import static io.trino.cost.StatsCalculator.noopStatsCalculator;
+import static io.trino.execution.warnings.WarningCollector.NOOP;
+import static io.trino.metadata.MetadataManager.createTestMetadataManager;
+import static io.trino.operator.scalar.ApplyFunction.APPLY_FUNCTION;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.testing.TestingEventListenerManager.emptyEventListenerManager;
+import static io.trino.testing.TestingSession.testSessionBuilder;
+import static io.trino.transaction.InMemoryTransactionManager.createTestTransactionManager;
+import static io.trino.transaction.TransactionBuilder.transaction;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.emptyMap;
+
+public class TestStatementRewrite
+{
+    private static final String TPCH_CATALOG = "tpch";
+    private static final CatalogName TPCH_CATALOG_NAME = new CatalogName(TPCH_CATALOG);
+    private static final Session.SessionBuilder CLIENT_SESSION_BUILDER = testSessionBuilder()
+            .setCatalog(TPCH_CATALOG)
+            .setSchema("s1")
+            .addPreparedStatement("q1", "SELECT a, ? as col2 FROM tpch.s1.t1");
+    private static final Session SETUP_SESSION = testSessionBuilder()
+            .setCatalog("c1")
+            .setSchema("s1")
+            .build();
+
+    private static final SqlParser SQL_PARSER = new SqlParser();
+    private static final DescribeOutputRewrite DESCRIBE_OUTPUT_REWRITE = new DescribeOutputRewrite();
+
+    private TransactionManager transactionManager;
+    private AccessControlManager accessControl;
+    private Metadata metadata;
+
+    @Test
+    public void testDescribeOutputFormatSql()
+    {
+        transaction(transactionManager, accessControl)
+                .readUncommitted()
+                .execute(beginTransaction(CLIENT_SESSION_BUILDER), session -> {
+                    Statement root = SQL_PARSER.createStatement("DESCRIBE OUTPUT q1", new ParsingOptions());
+                    Statement rewrittenNode = rewrite(session, root, DESCRIBE_OUTPUT_REWRITE);
+                    // SqlFormatterUtil#getFormattedSql will parse query and assert query round-trip inside.
+                    // We don't need to add extra asserting.
+                    SqlFormatterUtil.getFormattedSql(rewrittenNode, SQL_PARSER);
+                });
+    }
+
+    @Test
+    public void testDescribeInputFormatSql()
+    {
+        transaction(transactionManager, accessControl)
+                .readUncommitted()
+                .execute(beginTransaction(CLIENT_SESSION_BUILDER), session -> {
+                    Statement root = SQL_PARSER.createStatement("DESCRIBE INPUT q1", new ParsingOptions());
+                    Statement rewrittenNode = rewrite(session, root, DESCRIBE_OUTPUT_REWRITE);
```

Comment:
I guess this should be `DescribeInputRewrite`.

Reply:
oops, Yes, it should be `DescribeInputRewrite`. Thanks for the reminder.